connect-inject/container_init_test.go

```diff
@@ -68,6 +72,73 @@ consul-k8s connect-init -pod-name=${POD_NAME} -pod-namespace=${POD_NAMESPACE} \
 			"",
 		},
 
+		{
+			"When auth method is set and there is a service name annotation, -service-account-name is passed as an empty flag",
+			func(pod *corev1.Pod) *corev1.Pod {
+				pod.Annotations[annotationService] = "web"
+				pod.Spec.ServiceAccountName = "web"
+				pod.Spec.Containers[0].VolumeMounts = []corev1.VolumeMount{
+					{
+						Name:      "sa",
+						MountPath: "/var/run/secrets/kubernetes.io/serviceaccount",
+					},
+				}
+				return pod
+			},
+			Handler{
+				AuthMethod: "an-auth-method",
+			},
+			"",
+			`/bin/sh -ec 
+export CONSUL_HTTP_ADDR="${HOST_IP}:8500"
+export CONSUL_GRPC_ADDR="${HOST_IP}:8502"
+consul-k8s connect-init -pod-name=${POD_NAME} -pod-namespace=${POD_NAMESPACE} \
+  -acl-auth-method="an-auth-method" \
+`,
```

Comment:
Hmm as I'm reading this test name, I'm curious why is the `-service-account-name` not passed here?

Reply:
You are right it should have been! The tests have been changed now with the new approach you suggested